### PR TITLE
jpilot: init at 2.0.2

### PR DESCRIPTION
**Pull request by PapayaJackal**
Adds:
- [jpilot](https://github.com/juddmon/jpilot) - graphical tool for connecting to PalmOS handheld devices
- [pilot-link](https://github.com/desrod/pilot-link) - cli tools and library for connecting to PalmOS handheld devices
  -  dependency of jpilot

along with necessary patches to actually build these on modern systems

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/jp/jpilot/package.nix [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    sqlite
+  ];
+
+  preConfigure = ''
+    NOCONFIGURE=1 ./autogen.sh
+  '';
+
+  configureFlags = [ "--with-pilot-prefix=${pilot-link}" ];
+
+  meta = {
```
Comment (FliegendeWurst) [inline pkgs/by-name/jp/jpilot/package.nix:42]: Please fix darwin build error
```
       > ld: framework not found IOKit
```
or set platforms = linux.

File: pkgs/by-name/pi/pilot-link/package.nix [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+  bison,
+  enableLibpng ? true,
+  libpng,
+  enableLibusb ? true,
+  libusb-compat-0_1,
+}:
+
+stdenv.mkDerivation {
+  pname = "pilot-link";
+  version = "unstable-2022-09-26";
```
Comment (Aleksanaa) [inline pkgs/by-name/pi/pilot-link/package.nix:23]:
```suggestion
  version = "0_12_3-unstable-2022-09-26";
```

https://github.com/desrod/pilot-link/tags

File: pkgs/by-name/jp/jpilot/package.nix [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+    description = "Desktop organizer software for the Palm Pilot";
+    homepage = "https://www.jpilot.org/";
+    license = lib.licenses.gpl2;
+    maintainers = with lib.maintainers; [ PapayaJackal ];
```
Comment (Aleksanaa) [inline pkgs/by-name/jp/jpilot/package.nix:49]: missing `mainProgram`?

File: pkgs/by-name/jp/jpilot/package.nix [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+}:
+
+stdenv.mkDerivation {
+  pname = "jpilot";
+  version = "2.0.2";
+
+  src = fetchFromGitHub {
+    owner = "juddmon";
+    repo = "jpilot";
+    rev = "v2_0_2";
```
Comment (Aleksanaa) [inline pkgs/by-name/jp/jpilot/package.nix:21]:
```suggestion
    rev = "v${lib.replaceStrings [ "." ] [ "_" ] version}";
```

Add `rec` to `stdenv.mkDerivation`

File: pkgs/by-name/pi/pilot-link/package.nix (hunk 2 above)
Comment (FliegendeWurst) [reply pkgs/by-name/pi/pilot-link/package.nix:23]: Then you need to at least add "0-" in front.

**Pull request merged**